src/app/chip_data_model.gni

```diff
@@ -90,6 +90,7 @@ template("chip_data_model") {
       "${_app_root}/clusters/scenes/scenes-tokens.h",
       "${_app_root}/clusters/scenes/scenes.h",
       "${_app_root}/clusters/temperature-measurement-server/temperature-measurement-server.h",
+      "${_app_root}/clusters/thermostat-server/thermostat.cpp",
```

Comment:
This is not right.  This should be down in the `foreach(cluster, _cluster_sources) {` block below where the other "cpp name does not match dir name" cases are.

Reply:
Updated zap_cluster_list.py to include "thermostat-server"  
Also renamed thermostat-server/thermostat.cpp to thermostat-server/thermostat-server.cpp to avoid special case handling.